node-webservice/react-client/src/Components/GameplayPage.jsx

```diff
@@ -161,21 +140,62 @@ class GameplayPage extends Component {
         <div className='gameplay-page'>
           <h3>Playing game {this.props.match.params.gameid}</h3>
           <Grid onCellClick={this.handleCellClicked} parts={this.state.parts}></Grid>
+          <button onClick = {this.handleSubmit}> Submit </button>
+          <RobotJSONObjectForm/>
         </div>
       );
       //return <RobotJSONObjectForm />;
     }
     if (this.state.gameplayPhase === 'battle') {
+    
       return (
         <div className='gameplay-page'>
           <h3>Battle!</h3>
-          <Grid onCellClick={ () => {} } parts={this.state.parts}></Grid>          
+          <Grid onCellClick={ () => {} } parts={this.state.parts} gameplayPhase={this.state.gameplayPhase}></Grid>
+          <button onClick = {()=>this.changeBehaviour("target")}> Attack </button>
+          <button onClick = {()=>this.changeBehaviour("run")}> Defend </button>  
+          <button onClick = {()=>this.changeBehaviour("stand")}> Stand </button>            
         </div>
       );
     }
     return <ErrorPage>No page defined for game state: {this.state.gameplayPhase}</ErrorPage> 
   }
 
+  handleSubmit(event) {
+    event.preventDefault();
+
+    try {
+      socket.emit(
+        'game-message',
+        { action: "submitrobot", parts: this.state.parts, username: this.state.username},
+        response => {
+          if (response.error) {
+            alert("Error processing robot data: " + response.error);
+          }
+        }
+      );
+      //alert("Sent robot data");
+      this.setState({gameplayPhase:"battle"})
+    } catch(e) {
+      alert("Error sending robot data: " + e);
+    }
+  }
+
+  changeBehaviour(behaviour){
+    console.log(behaviour);
+    const behaviourMessage = {
+      action: "changeBehaviour",
+      username: this.state.username,
```

Comment:
the username in the message here doesn't hurt, but I don't think it's needed since the server will add it before emitting the message to other clients